core/src/main/scala/com/salesforce/op/evaluators/OpMultiClassificationEvaluator.scala

```diff
@@ -352,6 +539,54 @@ case class MultiClassificationMetricsTopK
   Error: Seq[Double]
 ) extends EvaluationMetrics
 
+/**
+ * Metrics for multi-class confusion matrix. It captures confusion matrix of records of which
+ * 1) the labels belong to the top n most occurring classes (n = confMatrixNumClasses)
+ * 2) the top predicted probability exceeds a certain threshold in confMatrixThresholds
+ *
+ * @param confMatrixNumClasses value of the top n most occurring classes in the dataset
+ * @param confMatrixClassIndices label index of the top n most occuring classes
+ * @param confMatrixThresholds a sequence of thresholds
+ * @param confMatrices a Matrix object that stores the confusion matrix for each threshold in confMatrixThresholds
+ */
+case class MulticlassConfMatrixMetricsByThreshold
+(
+  confMatrixNumClasses: Int,
```

Comment:
Don't you need to add `@JsonDeserialize` annotations here?

Reply:
This case class will be further transformed in E1 before saving the metrics, so there doesn't seem to be a need to add an annotation here. The JsonDeserialize tag was added to the Map as `ModelSelectorSummaryTest` will break without it.

Reply:
I think it's still a good idea to add deser annotations, in case someone would need to parse json into classes later on.

Reply:
@tovbinm added @JsonDeserialize annotation to both `ConfMatrixClassInidces` and `ConfMatrixThresholds` in the updated PR.